src/cider/nrepl/middleware/out.clj

```diff
@@ -18,6 +18,8 @@
 
 (declare unsubscribe-session)
 
+(set! *warn-on-reflection* true)
```

Comment:
We're using Eastwood as a reflection linter (CI-backed), you can remove this while also removing all of this https://github.com/clojure-emacs/cider-nrepl/blob/c3d6b69c98ef4c81529b69fa7624662f2006f545/project.clj#L150-L156

If the fix is indeed OK, Eastwood should be happy with it